src/components/Gallery/index.js

```diff
@@ -0,0 +1,188 @@
+}
+
+const Gallery = ({ params, minSelectionCount, maxSelectionCount, onSubmit, containerStyle }) => {
+  const [images, setImages] = useState(['Placeholder Key For TakePhotoButton'])
+  const [selectedImages, setSelectedImages] = useState([])
+  const [hasMoreImages, setHasMoreImages] = useState(true)
+  const [endCursor, setEndCursor] = useState(null)
+  const numColumns = (params) ? params.numColumns || DEFAULT_COLUMNS_COUNT : DEFAULT_COLUMNS_COUNT
+
+  const updateImagesAndCursor = (result) => {
+    const newImages = result.edges.map(obj => obj.node.image.uri).filter((image) => !images.includes(image))
+    setImages([...images, ...newImages])
+    setHasMoreImages(result.page_info.has_next_page)
+    setEndCursor(result.page_info.end_cursor)
+  }
+
+  const onSubmitPress = useCallback(
+    () => {
+      const length = selectedImages.length
+      if (length >= minSelectionCount && length <= maxSelectionCount) {
+        onSubmit(selectedImages)
+      }
+    },
+    [selectedImages, params]
+  )
+
+  const loadMore = useCallback(
+    async () => {
+      const granted = await requestReadPermissions()
+      if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
+        alertPermissionsError()
+        return
+      }
+
+      if (hasMoreImages) {
+        try {
+          updateImagesAndCursor(await getPhotos(endCursor))
+        } catch (error) {
+          alertPermissionsError()
+        }
+      }
+    },
+    [endCursor, hasMoreImages]
+  )
+
+  const canSelectMoreImages = useCallback(
+    () => selectedImages.length < maxSelectionCount,
+    [selectedImages]
+  )
+
+  const toggleSelect = useCallback(
+    (item) => {
+      const newSelectedImages = [...selectedImages]
+      const index = newSelectedImages.indexOf(item)
+
+      if (index !== -1) {
+        newSelectedImages.splice(index, 1)
+      } else {
+        if (canSelectMoreImages()) {
+          newSelectedImages.push(item)
+        }
+      }
+
+      setSelectedImages(newSelectedImages)
+    },
+    [selectedImages]
+  )
+
+  const callCamera = useCallback(
+    async () => {
+      const granted = await requestWritePermissions()
+
+      if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
+        alertPermissionsError()
+
+        return
+      }
+
+      try {
+        const [image] = await ImageService.openCamera()
+        const newImages = [...images]
+
+        newImages.splice(1, 0, image)
+        await CameraRoll.saveToCameraRoll(image)
+        setImages(newImages)
+      } catch (error) {
+      }
+    },
+    [images]
+  )
+
+  useEffect(() => {
+    if (!params) return
+
+    if (params.addedImages) {
+      const newImages = [...images]
+      newImages.splice(1, 0, ...params.addedImages)
+      setImages(newImages)
+    }
+
+    if (params.selectedImages) {
+      setSelectedImages([...params.selectedImages])
+    }
+  }, [params])
```

Comment:
Add `images` to the dependencies Array.

Reply:
This causes an infinite loop when `params.addedImage` is truthy, as `setImages` causes the hook to re-run

Reply:
@MPucheta then don't use the variable `images` because it can be outdated if it's not in the dependencies Array. To set a state based on a calculation from the previous value, you can pass a function:
```javascript
setImages(images => {
  const newImages = [...images]
  
  newImages.splice(1, 0, ...params.addedImages)

  return newImages
})
```